lib/Sema/TypeCheckMacros.cpp

```diff
@@ -1446,6 +1446,16 @@ swift::expandConformances(CustomAttr *attr, MacroDecl *macro,
     if (auto file = dyn_cast<FileUnit>(
             decl->getDeclContext()->getModuleScopeContext()))
       file->getOrCreateSynthesizedFile().addTopLevelDecl(extension);
+
+    // Most other macro-generated declarations are visited and type-checked
+    // through calling 'visitAuxiliaryDecls' on the original declaration the
+    // macro is attached to. We don't do this for macro-generated extensions,
+    // because the extension is not a peer of the original declaration. Instead
+    // of requiring all callers of 'visitAuxiliarDecls' to understsand the
+    // hoisting behavior of macro-generated extensions, we simply expose the
+    // extension to 'getTopLevelDecls()' and type check the extension eagerly
+    // here.
+    TypeChecker::typeCheckDecl(extension);
```

Comment:
Could we instead visit these at the end of TypeCheckDeclPrimary's visitation? Doing a complete check of the extension seems unnecessary here.

(Also, you have typos "visitAuxiliarDecls" and "understsand").

Reply:
I suppose it's better to visit expanded extensions in TypeCheckDeclPrimary as opposed to visiting the top-level decls of `SF->getSynthesizedFile()` at the end of `TypeCheckSourceFileRequest::evaluate` because the former approach will definitely invoke the request if it hasn't already happened?

Reply:
I hadn't thought of `TypeCheckSourceFileRequest::evaluate`. Visiting the top-level decls of the synthesized file there should work.